aten/src/ATen/native/LossCTC.cpp

```diff
@@ -397,17 +447,30 @@ Tensor ctc_loss(const Tensor& log_probs_, const Tensor& targets, IntArrayRef inp
     }
   }
   if (reduction == at::Reduction::Mean) {
-    auto target_lengths_t =
-        at::tensor(target_lengths, res.options()).clamp_min(1);
+    auto target_lengths_t = get_clamped_target_length(target_lengths, res.options());
     return (res / target_lengths_t).mean();
   } else if (reduction == at::Reduction::Sum) {
     return res.sum();
   }
   return is_batched ? res : res.squeeze(0);
 }
 
+} // namespace
+
+Tensor ctc_loss(const Tensor& log_probs_, const Tensor& targets, IntArrayRef input_lengths, IntArrayRef target_lengths, int64_t BLANK, int64_t reduction, bool zero_infinity) {
+  return ctc_loss_impl(log_probs_, targets, input_lengths, target_lengths, BLANK, reduction, zero_infinity);
+}
+
 // Convenience function accepting Tensors
 Tensor ctc_loss(const Tensor& log_probs, const Tensor& targets, const Tensor& input_lengths, const Tensor& target_lengths, int64_t BLANK, int64_t reduction, bool zero_infinity) {
+  if (at::areAnyTensorSubclassLike(
+          {log_probs, targets, input_lengths, target_lengths})) {
```

Comment:
Hey @zou3519 - `ctc_loss()` is an example of a "non-composite compliant decomposition", where it looks like we work around it by adding a composite compliant path that only executes some of the time.

Here, it looks like we only run it when any inputs are tensor subclasses. But, I think we want this path to also active wait:
- any transforms are active (specifically functionalization, and probably the other functorch transforms
- any python modes are active (maybe that's also checked by `areAnyTensorSubclassLike`)


Do you agree with that? Should we update `areAnyTensorSubclassLike()` (and/or rename it) to properly reflect that criteria?

That should fix https://github.com/pytorch/pytorch/issues/86384

Reply:
I thought this path already is active on functorch transforms, but I am not sure about the interaction with python modes. IMO `areAnyTensorSubclassLike` should cover all functorch transforms and Python Modes

Reply:
Oh maybe this is the bug. This keyset should probably include `DispatchKey::Functionalize`, right? I can make a patch

https://github.com/pytorch/pytorch/blob/bc919ac7963be9f113b4e3c8b668905404301f8f/aten/src/ATen/TensorSubclassLikeUtils.h#L27

Reply:
Yeah that looks like the bug, good find